### PR TITLE
Added LFU Cache

### PR DESCRIPTION
**Pull request by ruppysuppy**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
The comment said that None was returned but the failing test demonstrates that is not the case.

**Comment by ruppysuppy** [comment]
@cclauss, when None is returned in python interpreter or doctests, its not shown. For example try this code in the interpreter:
```
>>> def f():
	return None

>>> f()
>>> 
```
Even though f returns None, running f() won't display None in the interpreter or doctests

File: other/lfu_cache.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+from typing import Callable, Optional
+
+
+class DoubleLinkedListNode:
+    '''
```
Comment (mateuszz0000) [inline other/lfu_cache.py:5]: Please change `'''` to `"""` in docstrings

File: other/lfu_cache.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        '''
+
+        temp = self.rear.prev
+
+        self.rear.prev, node.next = node, self.rear
+        temp.next, node.prev = node, temp
+        node.freq += 1
+        self._position_node(node)
+
+    def _position_node(self, node):
```
Comment (mateuszz0000) [inline other/lfu_cache.py:39]: Missing typehint

File: other/lfu_cache.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+        return node
+
+
+class LFUCache:
+    '''
+    LFU Cache to store a given capacity of data. Can be used as a stand-alone object
+    or as a function decorator.
+
+    >>> cache = LFUCache(2)
```
Comment (mateuszz0000) [inline other/lfu_cache.py:63]: Could `LFUCache` take optional cache parameter? Would be easier testable.

```
 def __init__(self, capacity: int, cache: dict = None):
        self.list = DoubleLinkedList()
        self.capacity = capacity
        self.num_keys = 0
        self.hits = 0
        self.miss = 0
        self.cache = cache or {}
```

File: other/lfu_cache.py (hunk 1 above)
Comment (cclauss) [reply other/lfu_cache.py:5]: autoblack will do this automatically so let's not slow the review process on this.

File: other/lfu_cache.py (hunk 3 above)
Comment (cclauss) [reply other/lfu_cache.py:63]: That seems like a big change.

**Pull request merged**